iofog-agent-daemon/src/main/java/org/eclipse/iofog/process_manager/DockerUtil.java

```diff
@@ -185,8 +187,22 @@ public void startContainer(Microservice microservice) throws NotFoundException,
 //
 //		if (totalMemory - jvmMemory < requiredMemory)
 //			throw new Exception("Not enough memory to start the container");
-    	LoggingService.logDebug(MODULE_NAME , "start Container");
-        dockerClient.startContainerCmd(microservice.getContainerId()).exec();
+        try {
+            LoggingService.logDebug(MODULE_NAME , "start Container");
```

Comment:
Why catch all 3 exceptions when the logic for each is the same?

Reply:
I'll update the error messages. I think we should always catch the checked exception and the exception in the end.

Reply:
If you are just logging the exception message, you only need to catch the most general exception. You don't have any exception-specific behaviour